Tests/UnitTests/Plugins/DD4hep/DD4hepCylindricalDetectorTests.cpp

```diff
@@ -0,0 +1,400 @@
+                <layer id="0"  name="BeamPipeLayer">                 
+)"""";
+
+const char* nec_head_xml =
+    R""""(
+        <detector id="1" name="PixelNegativeEndcap" type="EndcapDetector" readout="PixelReadout">
+            <type_flags type="DetType_TRACKER + DetType_BARREL"/>
+)"""";
+
+const char* barrel_head_xml =
+    R""""(
+        <detector id="2" name="PixelBarrel" type="BarrelDetector" readout="PixelReadout">
+            <type_flags type="DetType_TRACKER + DetType_BARREL"/>
+)"""";
+
+const char* pec_head_xml =
+    R""""(
+        <detector id="3" name="PixelPositiveEndcap" type="EndcapDetector" readout="PixelReadout">
+            <type_flags type="DetType_TRACKER + DetType_BARREL"/>
+)"""";
+
+const char* plugin_xml =
+    R"""(<plugins>
+    <plugin name="DD4hep_ParametersPlugin">
+      <argument value="world"/>
+      <argument value="acts_world: bool = true"/>
+      <argument value="acts_world_type: int = 3"/>
+      <argument value="acts_world_bvalues_n: int = 3"/>
+      <argument value="acts_world_bvalues_0: double = 0"/>
+      <argument value="acts_world_bvalues_1: double = 150*mm"/>
+      <argument value="acts_world_bvalues_2: double = 1000*mm"/>
+      <argument value="acts_world_binning: str = r"/>
+      <argument value="acts_world_geo_id: str = incremental"/>
+      <argument value="acts_world_root_volume_finder: str = indexed"/>
+    </plugin>	
+    <plugin name="DD4hep_ParametersPlugin">
+      <argument value="BeamPipe"/>
+      <argument value="acts_volume: bool = true"/>
+      <argument value="acts_volume_type: int = 3"/>
+      <argument value="acts_volume_bvalues_n: int = 3"/>
+      <argument value="acts_volume_bvalues_0: double = 0"/>
+      <argument value="acts_volume_bvalues_1: double = 20*mm"/>
+      <argument value="acts_volume_bvalues_2: double = 1000*mm"/>
+      <argument value="acts_volume_internals: bool = true"/>
+      <argument value="acts_volume_internals_type: str = layer"/>
+    </plugin>	
+    <plugin name="DD4hep_ParametersPlugin">
+      <argument value="Pixel"/>
+      <argument value="acts_container: bool = true"/>
+      <argument value="acts_container_type: int = 3"/>
+      <argument value="acts_container_bvalues_n: int = 3"/>
+      <argument value="acts_container_bvalues_0: double = 20*mm"/>
+      <argument value="acts_container_bvalues_1: double = 150*mm"/>
+      <argument value="acts_container_bvalues_2: double = 1000*mm"/>
+      <argument value="acts_container_binning: str = z"/>
+    </plugin>	
+    <plugin name="DD4hep_ParametersPlugin">
+      <argument value="PixelNegativeEndcap"/>
+      <argument value="acts_container: bool = true"/>
+      <argument value="acts_container_type: int = 3"/>
+      <argument value="acts_container_bvalues_n: int = 3"/>
+      <argument value="acts_container_bvalues_0: double = 20*mm"/>
+      <argument value="acts_container_bvalues_1: double = 150*mm"/>
+      <argument value="acts_container_bvalues_2: double = 200*mm"/>
```

Comment:
Do you think it's be worth to add vector support to the parameter plugin on the DD4hep side to try to make this simpler?

Reply:
I have worked around it now, but yes, `vector<double>` and `vector<int>` is something I could use there.